woodchipper/http/fastapi.py

```diff
@@ -1,80 +1,84 @@
+                        "http.response.content_length": int(dict(message["headers"]).get(b"content-length", 0)),
+                    }
+                )
+            return await send(message)
 
-                with LoggingContext(
-                    "fastapi:request",
-                    **{
-                        "id": gen_id(),
-                        "body_size": int(request.headers.get("content-length", 0)),
-                        "method": request.method,
-                        "path": request.base_url._url,
-                        **{f"query_param.{k.lower()}": v for k, v in queries.items()},
-                        **{
-                            f"header.{k.lower()}": (v if k.lower() not in blacklisted_headers else "******")
-                            for k, v in request.headers.items()
-                        },
-                    },
-                    _prefix="http",
-                ):
-                    try:
-                        response: Response = await original_route_handler(request)
-                    except Exception as e:
-                        logging_ctx.update({"http.response.status_code": 500})
-                        raise e
-                    else:
-                        logging_ctx.update(
-                            {
-                                "http.response.status_code": response.status_code,
-                                "http.response.content_length": int(response.headers.get("content-length", 0)),
-                            }
-                        )
-                        return response
-
-            return woodchipper_route_handler
-
-        return get_route_handler
-
-    def get_logging_route_class(self):
-        return type(
-            "WoodChipperApiRoute",
-            (
-                self._app.router.route_class,
-            ),  # This allows us to handle cases where the routeclass has already been replaced
-            {"get_route_handler": self.get_custom_route_handler(self._request_id_factory, self._blacklisted_headers)},
-        )
+        with LoggingContext(
+            "fastapi:request",
+            **{
+                "id": self._request_id_factory(),
+                "body_size": int(request.headers.get("content-length", 0)),
+                "method": request.method,
+                "path": request.base_url._url,
+                **{f"query_param.{k.lower()}": v for k, v in queries.items()},
+                **{
+                    f"header.{k.lower()}": (v if k.lower() not in self._blacklisted_headers else "******")
+                    for k, v in request.headers.items()
+                },
+            },
+            _prefix="http",
+        ):
+            try:
+                await self._app(scope, receive, send_with_extra_headers)
+            except Exception as e:
+                logging_ctx.update({"http.response.status_code": 500})
+                raise e
```

Comment:
[sand] You want just plain `raise` here.

Reply:
This will be any uncaught error. It will ultimately be reported as a 500.

Reply:
Thanks, fixing!